azure-cosmos/src/main/java/io/micronaut/azure/cosmos/client/CosmosClientFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.azure.cosmos.client;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosClient;
+import io.micronaut.context.annotation.Bean;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Requires;
+
+/**
+ * The Azure Cosmos Client factory.
+ *
+ * @author radovanradic
+ * @since TODO
+ */
+@Factory
+public final class CosmosClientFactory {
+
+    @Bean(preDestroy = "close")
+    @Requires(beans = CosmosClientConfiguration.class)
+    CosmosClient buildCosmosClient(CosmosClientConfiguration configuration) {
```

Comment:
add javadoc

Reply:
Also, if I'm not mistaken, `@Bean` will create a prototype bean. You need to add `@Singleton` too.

Reply:
`@Factory` is defined with `@DefaultScope(Singleton.class)`

Reply:
that is for the factory itself and not the beans it creates

Reply:
@radovanradic could you add `@Singleton` to each of the beans please

Reply:
Updated